src/librbd/cache/ImageCache.h

```diff
@@ -44,10 +51,10 @@ struct ImageCache {
   /// internal state methods
   virtual void init(Context *on_finish) = 0;
   virtual void shut_down(Context *on_finish) = 0;
+  virtual void get_state(bool &clean, bool &empty, bool &present) = 0;
 
-  virtual void invalidate(Context *on_finish) = 0;
+  virtual void invalidate(Context *on_finish, bool discard_unflushed_writes=false) = 0;
```

Comment:
What's the purpose of this new ```discard_unflushed_writes``` parameter? It doesn't seem to be used in this patchset?

Reply:
The ImageCache.h interface has just invalidate(*ctx). We override that here, and the idea was to map that to our variant of invalidate with the discard flag. For some reason I thought it made more  sense at the time to define the "does invalidate discard unflushed writes or not" behavior there, rather than in RWL::invalidate().

It doesn't seem simpler now given how long it just took me to recall all this. On top of that the path from calling RWL::invalidate(ctx) to RWL::invalidate(ctx, false) seems unclear.

I really needed all that overloading with RWL::flush() and may have just gone overboard with that.